main.go

```diff
@@ -261,24 +261,17 @@ func route(w dns.ResponseWriter, req *dns.Msg) {
 
 	log.WithFields(log.Fields{"question": fqdn, "type": rrString, "client": clientIp, "proto": proto}).Debug("Request")
 
-	if msg := clientSpecificCacheHit(clientIp, req); msg != nil {
+	if msg, exp := clientSpecificCacheHit(clientIp, req); msg != nil {
 		if len(msg.Answer) > 1 {
```

Comment:
@kinarashah Looks like the code block 265-271 can be extracted to a separate method as it is being repeated in the file several times